File: application/src/main/java/com/github/dkurata38/open_web_library/domain/member/Member.java
package com.github.dkurata38.open_web_library.domain.member;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

import static java.util.stream.Collectors.*;

@RequiredArgsConstructor
public class Member {
	@NotNull
	@Getter
	private final MemberId memberId;
	@NotEmpty
	@Getter
	private final String name;
}

File: member-domain/src/main/kotlin/com/github/dkurata38/open_web_library/member_domain/Member.kt
package com.github.dkurata38.open_web_library.member_domain

import java.time.LocalDateTime

data class Member(val id: MemberId, val name: String, val registeredAt: LocalDateTime) {
}

data class MemberId(val value: String) {
	companion object {
		fun issue(): MemberId {
			return MemberId(IdIssuer.generateIdValue())
		}
	}
}
